packages/server/controllers/auth.js

```diff
@@ -102,9 +104,35 @@ async function deserialize(id, callback) {
 
 async function authenticate(request, response, next) {
   const authToken = request.get("Authorization");
-  const payload = token.verify(authToken);
-  request.user = payload;
-  next();
+  
+  try{
+  token.verifyAccessToken(authToken);
+
+  jwt.verify(authToken, process.env.WEB_CLIENT_SECRET, function(err, decoded) {
+    if (err) {
+      if (err.name === 'TokenExpiredError') {
+        // Token has expired
+        response.status(401).json({ 
+          error: 'Access token has expired' 
+        });
+      } else {
+        // Other token verification errors
+        response.status(401).json({ 
+          error: 'Invalid access token' 
+        });
+      }
+    } else {
+      // Token is valid
+      request.user = decoded;
+      next();
+    }
+  });
```

Comment:
Why do you have `token.verifyAccessToken(authToken);` and then verify the token again in the next line? You can move this logic to `verifyAccessToken` and throw errors where the `response.status().json()` methods are. That way, the errors you throw in `verifyAccessToken` will be caught and handled in this try/catch block instead.